source/ui/memory-component-presenter.cpp

```diff
@@ -237,38 +206,46 @@ MemoryValue MemoryComponentPresenter::getMemoryValueCached(
     // set cache status to valid
     _memoryCacheValid = true;
 
-    std::size_t subset_start = (memory_address - _memoryCacheBaseAddress) * 8;
-    std::size_t subset_end =
-        (memory_address - _memoryCacheBaseAddress + memory_length) * 8;
-    return _memoryCache.subSet(subset_start, subset_end);
+    std::size_t subsetStart = (memoryAddress - _memoryCacheBaseAddress) * 8;
+    std::size_t subsetEnd =
+        (memoryAddress - _memoryCacheBaseAddress + memoryLength) * 8;
+    return _memoryCache.subSet(subsetStart, subsetEnd);
   }
 }
 
 
 QHash<int, QByteArray> MemoryComponentPresenter::roleNames() const {
-  // connect TableColumns in View with columns in this model
-  QHash<int, QByteArray> roles;
-  roles[AddressRole] = "address";
-  roles[ValueRoleBin8] = "bin8";
-  roles[ValueRoleBin16] = "bin16";
-  roles[ValueRoleBin32] = "bin32";
-  roles[ValueRoleBin64] = "bin64";
-  roles[ValueRoleOct8] = "oct8";
-  roles[ValueRoleOct16] = "oct16";
-  roles[ValueRoleOct32] = "oct32";
-  roles[ValueRoleOct64] = "oct64";
-  roles[ValueRoleHex8] = "hex8";
-  roles[ValueRoleHex16] = "hex16";
-  roles[ValueRoleHex32] = "hex32";
-  roles[ValueRoleHex64] = "hex64";
-  roles[ValueRoleDec8] = "dec8";
-  roles[ValueRoleDec16] = "dec16";
-  roles[ValueRoleDec32] = "dec32";
-  roles[ValueRoleDec64] = "dec64";
-  roles[ValueRoleDecS8] = "decs8";
-  roles[ValueRoleDecS16] = "decs16";
-  roles[ValueRoleDecS32] = "decs32";
-  roles[ValueRoleDecS64] = "decs64";
-  roles[InfoRole] = "info";
+  // clang-format off
+  static QHash<int, QByteArray> roles = {
+    {AddressRole, "address"},
+    {ValueRoleBin8, "BinaryData8"},
+    {ValueRoleBin16, "BinaryData16"},
+    {ValueRoleBin32, "BinaryData32"},
+    {ValueRoleBin64, "BinaryData64"},
+    {ValueRoleHex8, "HexData8"},
+    {ValueRoleHex16, "HexData16"},
+    {ValueRoleHex32, "HexData32"},
+    {ValueRoleHex64, "HexData64"},
+    {ValueRoleDec8, "UnsignedDecimalData8"},
+    {ValueRoleDec16, "UnsignedDecimalData16"},
+    {ValueRoleDec32, "UnsignedDecimalData32"},
+    {ValueRoleDec64, "UnsignedDecimalData64"},
+    {ValueRoleDecS8, "SignedDecimalData8"},
+    {ValueRoleDecS16, "SignedDecimalData16"},
+    {ValueRoleDecS32, "SignedDecimalData32"},
+    {ValueRoleDecS64, "SignedDecimalData64"},
+    {InfoRole, "info"}
+  };
+  // clang-format on
   return roles;
 }
+
+QString MemoryComponentPresenter::_roleToDataFormat(const QString &role) {
+  QString dataFormat = role;
+  if (dataFormat.endsWith("8")) {
```

Comment:
What happens for `BinaryData128`? Better use a regular expression. `QString` has [nice support for this](http://doc.qt.io/qt-5/qstring.html#remove-4):

```
// Take by value
QString MemoryComponentPresenter::_roleToDataFormat(QString role) {
  return role.remove(QRegularExpression("\\d+$"));
}
```
something like that